seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/maxwell/MaxWellJsonDeserializationSchema.java

```diff
@@ -0,0 +1,230 @@
+        this.database = database;
+        this.table = table;
+        this.fieldNames = physicalRowType.getFieldNames();
+        this.fieldCount = physicalRowType.getTotalFields();
+        this.ignoreParseErrors = ignoreParseErrors;
+        this.databasePattern = database == null ? null : Pattern.compile(database);
+        this.tablePattern = table == null ? null : Pattern.compile(table);
+    }
+
+    @Override
+    public SeaTunnelRow deserialize(byte[] message) throws IOException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SeaTunnelDataType<SeaTunnelRow> getProducedType() {
+        return this.physicalRowType;
+    }
+
+    @Override
+    public void deserialize(byte[] message, Collector<SeaTunnelRow> out) {
+        if (message == null) {
+            return;
+        }
+        ObjectNode jsonNode = (ObjectNode) convertBytes(message);
+        assert jsonNode != null;
+        if (database != null
+                && !databasePattern.matcher(jsonNode.get(FIELD_DATABASE).asText()).matches()) {
+            return;
+        }
+        if (table != null && !tablePattern.matcher(jsonNode.get(FIELD_TABLE).asText()).matches()) {
+            return;
+        }
+        JsonNode dataNode = jsonNode.get(FIELD_DATA);
+        String type = jsonNode.get(FIELD_TYPE).asText();
+        if (OP_INSERT.equals(type)) {
+            SeaTunnelRow rowInsert = convertJsonNode(dataNode);
+            rowInsert.setRowKind(RowKind.INSERT);
+            out.collect(rowInsert);
+        } else if (OP_UPDATE.equals(type)) {
+            SeaTunnelRow rowAfter = convertJsonNode(dataNode);
+            JsonNode oldNode = jsonNode.get(FIELD_OLD);
+            SeaTunnelRow rowBefore = convertJsonNode(oldNode);
+            for (int f = 0; f < fieldCount; f++) {
+                assert rowBefore != null;
+                if (rowBefore.isNullAt(f) && oldNode.findValue(fieldNames[f]) == null) {
+                    // fields in "old" (before) means the fields are changed
+                    // fields not in "old" (before) means the fields are not changed
+                    // so we just copy the not changed fields into before
+                    assert rowAfter != null;
+                    rowBefore.setField(f, rowAfter.getField(f));
+                }
+            }
+            assert rowBefore != null;
+            rowBefore.setRowKind(RowKind.UPDATE_BEFORE);
+            assert rowAfter != null;
+            rowAfter.setRowKind(RowKind.UPDATE_AFTER);
+            out.collect(rowBefore);
+            out.collect(rowAfter);
+        } else if (OP_DELETE.equals(type)) {
+            SeaTunnelRow rowDelete = convertJsonNode(dataNode);
+            rowDelete.setRowKind(RowKind.DELETE);
+            out.collect(rowDelete);
+        } else {
+            if (!ignoreParseErrors) {
+                throw new SeaTunnelJsonFormatException(
+                        CommonErrorCodeDeprecated.UNSUPPORTED_DATA_TYPE,
```

Comment:
`CommonErrorCodeDeprecated` has been annotated with `@Deprecated`. Please replace it with `CommonErrorCode`.